### ISSUE TITLE
Bug: Helix Authentication Service Configuration isn't persistent  

### ISSUE THREAD

**Issue by keith-miller** [aws-games/cloud-game-development-toolkit#382]
### Expected Behaviour

Whenever the Helix Authentication Service container restarts, the stored configurations should persist and be available when the service comes back up.

### Current Behaviour

Whenever the Helix Authentication Service container restarts, the stored configurations are lost and have to be reentered. 

### Possible Solution

This I have no idea about. I don't see anything in their documentation about storage or where the application could be storing the configurations. 

### Steps to Reproduce

1. Create the service
2. Add a configuration 
3. Restart the service
4. Check for the configuration

### Cloud Game Development Toolkit version

latest

**Comment by kylesomers** [comment]
Thanks for opening this issue @keith-miller ! How were you modifying the configuration? 

Helix Authentication Service provides the ability to set configuration using system-level environment variables, so our Helix Auth Terraform module sets them as environment variables defined in the ECS Task Definition (see [here](https://github.com/aws-games/cloud-game-development-toolkit/blob/c687f037814d86428f7a3aa696b8991b29ad7d6f/modules/perforce/helix-authentication-service/main.tf#L77)). These are applied when the module is deployed. If you make any configurations to a running Helix Auth server (deployed ECS Task) outside of this mechanism, they won't persist across restarts. This behavior was intended but probably warrants more explanation in our module docs, which I'll make sure is updated. If you want to make Helix Auth configuration changes, we recommend you do so in the Terraform module in that env var block, where you can replace values or add new environment variables as you need. This way the change can be managed as part of the state of your terraform configuration, and provides a path for these kinds of changes to be managed through an automated CI/CD workflow (if you use one). ECS default behavior does a rolling deploy to replace the running container with a new one that has the updated configuration.

**Comment by henrykie** [comment]
@kylesomers Configuring OIDC through admin portal would not persist across restarts though - thats the problem here. As long as we enable the admin endpoint a restart of the task doesn't propagate. This could be adjusted if we backed up the .env (or wherever OIDC configurations are set by the admin portal)to a persistent store.

**Comment by kylesomers** [comment]
I propose we don't include the admin portal, unless we get feedback that its really useful. Backing up and maintaining synced state of those changes adds more failure risk.

**Comment by keith-miller** [comment]
@henrykie that's correct, after install we used the admin panel to configure. After container restarts those changes are lost. As containers are ephemeral we'd like a way to make the configuration consistent.

**Comment by henrykie** [comment]
> If that configuration can be kept in TF, would that make administration easier?

I think the fix that we proposed would see that configuration being stored in a secret, so your TF config would just reflect the ARN of that secret.

**Comment by henrykie** [comment]
**New Approach**

Due to the limitations identified by @kylesomers with the shared bind mount between the sidecar and service container, we are unable to inject the `config.toml` prior to Helix Authentication Service startup. We've proposed shifting to EC2 Image Builder for a custom HAS docker image being built prior to service launch. We'll take this up during our next sprint.
